go/def.bzl

```diff
@@ -575,18 +534,7 @@ def go_test_impl(ctx):
   )
 
 go_env_attrs = {
-    "toolchain": attr.label(
-        default = Label("//go/toolchain:toolchain"),
-        allow_files = True,
-        cfg = "host",
-    ),
-    "go_tool": attr.label(
-        default = Label("//go/toolchain:go_tool"),
-        single_file = True,
-        allow_files = True,
-        executable = True,
-        cfg = "host",
-    ),
+    "go_toolchain": attr.label(default = Label("@io_bazel_rules_go_toolchain//:go_toolchain")),
```

Comment:
I assume users won't be setting go_toolchain or toolchain attributes on their rules? If so, I'd recommend using _go_toolchain and _toolchain for the attribute names to indicate they are private to the rule implementation.

Also, add providers = go_toolchain_type to the arguments to attr.label.